packages/spatial_pooler/src/nupic/research/frameworks/spatial_pooler/spatial_pooler.py

```diff
@@ -0,0 +1,834 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2022, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+import numpy as np
+import itertools
+
+real_type = np.float32
+uint_type = np.uint32
+
+class SpatialPooler():
```

Comment:
Add comments that explain what the spatial pooler is to a naive reader. Simplest is to have one or two sentences explaining it and a second sentence pointing to the SP paper.

Reply:
Done.